src/main/java/com/transloadit/sdk/Assembly.java

```
@@ -1,64 +1,100 @@
+    private TusURLStore tusURLStore;
+
     protected Map<String, File> files;
+    protected Map<String, InputStream> fileStreams;
     protected TusClient tusClient;
+    protected List<TusUpload> uploads;
 
     public Assembly(Transloadit transloadit) {
         this(transloadit, new Steps(), new HashMap<String, File>(), new HashMap<String, Object>());
     }
 
     /**
-     *
      * @param transloadit {@link Transloadit} the transloadit client.
-     * @param steps {@link Steps} the steps to add to the assembly.
-     * @param files is a map of file names and files that are meant to be uploaded.
-     * @param options map of extra options to be sent along with the request.
+     * @param steps       {@link Steps} the steps to add to the assembly.
+     * @param files       is a map of file names and files that are meant to be uploaded.
+     * @param options     map of extra options to be sent along with the request.
      */
     public Assembly(Transloadit transloadit, Steps steps, Map<String, File> files, Map<String, Object> options) {
         this.transloadit = transloadit;
         this.steps = steps;
         this.files = files;
         this.options = options;
-        tusClient = null;
+        tusClient = new TusClient();
+        tusURLStore = new TusURLMemoryStore();
+        uploads = new ArrayList<TusUpload>();
+        fileStreams = new HashMap<String, InputStream>();
     }
 
     /**
      * Adds a file to your assembly.
      *
      * @param file {@link File} the file to be uploaded.
-     * @param name {@link String} the name you the file to be given in transloadit
+     * @param name {@link String} the name you the file to be given in transloadit.
      */
-    public void addFile(File file, String name){
+    public void addFile(File file, String name) {
         files.put(name, file);
+
+        // remove duplicate key
+        if (fileStreams.containsKey(name)) {
+            fileStreams.remove(name);
```

What about being explicit and throwing an exception here instead of silently removing the first file stream?

---

@Acconut I am removing it silently to keep the old behaviour we had from the `map.put`, which would override previously added files with that same name.

---

Ok, makes sense. Would you mind adding this to the method's JavaDoc.

---

👍

---

@Acconut added, thanks